app/components/wallet/trezorConnect/SaveDialog.js

```diff
@@ -0,0 +1,189 @@
+    defaultMessage: '!!!Save',
+    description: 'Label for the "Save" button on the Connect to Trezor Hardware Wallet dialog.'
+  },
+});
+
+messages.fieldIsRequired = globalMessages.fieldIsRequired;
+
+type Props = {
+  progressInfo: ProgressInfo,
+  error: ?LocalizableError,
+  isActionProcessing: boolean,
+  defaultWalletName: string,
+  submit: Function,
+  cancel: Function,
+};
+
+@observer
+export default class SaveDialog extends Component<Props> {
+
+  static contextTypes = {
+    intl: intlShape.isRequired
+  };
+
+  // form for wallet name
+  form: typeof ReactToolboxMobxForm;
+
+  componentWillMount() {
+    const { intl } = this.context;
+
+    this.form = new ReactToolboxMobxForm({
+      fields: {
+        walletName: {
+          label: intl.formatMessage(messages.saveWalletNameInputLabel),
+          placeholder: intl.formatMessage(messages.saveWalletNameInputPlaceholder),
+          value: this.props.defaultWalletName,
+          validators: [({ field }) => (
+            [
+              isValidHardwareWalletName(field.value),
+              intl.formatMessage(globalMessages.invalidHardwareWalletName)
+            ]
+          )],
+        },
+      },
+    }, {
+      options: {
+        validateOnChange: true,
+        validationDebounceWait: 250,
+      },
+    });
+  }
+
+  render() {
+    const { intl } = this.context;
+
+    // walletNameBlock
+    const walletNameFieldClasses = classnames([
+      'walletName',
+      styles.walletName,
+    ]);
+    const walletNameField = this.form.$('walletName');
+
+    const walletNameBlock = (
+      <div className={classnames([styles.headerBlock, styles.headerSaveBlock])}>
+        <Input
+          className={walletNameFieldClasses}
+          {...walletNameField.bind()}
+          error={walletNameField.error}
+          skin={<SimpleInputSkin />}
+        />
+        <span>{intl.formatMessage(messages.saveWalletNameInputBottomInfo)}</span>
+      </div>);
+
+    // middleBlock selection depending upon state
+    let middleBlock = null;
+
+    switch (this.props.progressInfo.stepState) {
+      case StepStateOption.LOAD:
+        middleBlock = (
+          <div className={classnames([styles.middleBlock, styles.middleSaveLoadBlock])}>
+            <SvgInline svg={saveLoadGIF} cleanup={['title']} />
+          </div>);
+        break;
+      case StepStateOption.PROCESS:
+        // START
+        middleBlock = (
+          <div className={classnames([styles.middleBlock, styles.middleSaveStartProcessBlock])}>
+            <SvgInline svg={saveStartSVG} cleanup={['title']} />
+          </div>);
+        break;
+      case StepStateOption.ERROR:
+        middleBlock = (
+          <div className={classnames([styles.middleBlock, styles.middleSaveErrorBlock])}>
+            <SvgInline svg={saveErrorSVG} cleanup={['title']} />
+          </div>);
+        break;
+      default:
+        console.error('Error : something unexpected happened');
```

Comment:
change to `Logger`. I would try to always keep the `:` right after any text that is before (without the space)

Reply:
DONE